wiki/Manual.md

```diff
@@ -2694,6 +2695,47 @@ Returns
 }
 ```
 
+## Open Interest History
+
+*contract only*
+
+Use the `fetchOpenInterestHistory` method to get a history of open interest for a symbol from the exchange.
+
+```JavaScript
+fetchOpenInterestHistory (symbol, timeframe = '5m', since = undefined, limit = undefined, params = {})
+```
+
+Parameters
+
+- **symbol** (String) Unified CCXT market symbol (e.g. `"BTC/USDT:USDT"`)
+- **timeframe** (String) Check exchange.timeframes for available values
+- **since** (Integer) Timestamp for the earliest open interest record (e.g. `1645807945000`)
+- **limit** (Integer) The maximum number of [open interest structures](#open-interest-structures) to retrieve (e.g. `10`)
+- **params** (Dictionary) Extra parameters specific to the exchange API endpoint (e.g. `{"endTime": 1645807945000}`)
+
+Returns
+
+- An array of [open interest structures](#open-interest-structure)
+
+### Open Interest Structure
+
+```JavaScript
+{
+    symbol: 'BTC/USDT',
+    volume: 80872.801,
+    value: 3508262107.38,
+    valueCurrency: 'USDT',
```

Comment:
Can we rename this from `valueCurrency` to `currency` and comment on the value and the volume currencies in the docs ? I would rather use `baseVolume` and `quoteVolume` here instead for consistency with the terms we use in tickers.

Reply:
Look at the Okex PR, it takes a currency code instead of a market symbol, so thats the only potential problem I see

Reply:
What is the asset of `volume` ?

_sidenote: let's put a proper contract symbol in the docs, cause open interest does not apply to spot markets_

Reply:
> What is the asset of `volume` ? Is it in base currency or in contracts ?

I honestly don't know, it might depend on the exchange, I was just returning the value that was sent as volume

Reply:
> I honestly don't know, it might depend on the exchange, I was just returning the value that was sent as volume

Can you write out the list of exchanges having it and investigate which asset is meant in each case ?

Reply:
> sidenote: let's put a proper contract symbol in the docs, cause open interest does not apply to spot markets

What do you mean, it's labelled as *contract only* already

![](https://i.imgur.com/32Ur9wm.png)

Reply:
> > I honestly don't know, it might depend on the exchange, I was just returning the value that was sent as volume
> 
> Can you write out the list of exchanges having it and investigate which asset is meant in each case ? If it's always the base volume, then we should use one naming schema, if it's different from exchange to exchange then we have to add a currency field for it.

Base and quote are always the same, but i don't know if it's contracts, and on okex you provide a currency code instead of a market symbol, so there is no base and quote, but there is a volume and a value still

Reply:
I mean despite that it's labeled _contract-only_ the symbol in the documented structure is a spot-symbol. That's a typo in the docs.